File: zig/pangram/pangram.zig
const std = @import("std");

const NUM_ALPHABETS: u16 = 26;

pub fn isPangram(str: []const u8) bool {
    var letters = std.bit_set.IntegerBitSet(NUM_ALPHABETS).initEmpty();

    for (str) |c| {
        if (!std.ascii.isAlphabetic(c)) continue;
        letters.set(std.ascii.toLower(c) - 'a');
    }

    return letters.count() == NUM_ALPHABETS;
}

File: cpp/pangram/pangram.cpp
#include "pangram.h"

#include <bitset>

namespace pangram {
bool is_pangram(std::string s) {
    std::bitset<NUM_ALPHABETS> letters;

    for (const auto c : s) {
        if (std::isalpha(c)) letters.set(std::tolower(c) - 'a');
    }

    return letters.all();
}
}  // namespace pangram
